src/goto-programs/README.md

```diff
@@ -509,3 +509,80 @@ C++ modules:
 Dependencies:
   - [linking folder](\ref linking).
   - [typecheck](\ref section-goto-typecheck).
+
+\section java-trace-structure Java Trace Structure
```

Comment:
I'm not sure I agree that talking about Java should happen in this folder. Looking at the contents, however, it doesn't particularly Java-specific anyway. Can the title and the introduction below be adjusted a bit to make clear that it's only about the example being in Java, which seems perfectly fine to me?